File: junit-service/src/main/java/com/learn/java/TestDemo.java
package com.learn.java;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * 描述信息
 *
 * @create: 2022-08-09 19:55
 */
public class TestDemo {

    @Before
    public void before(){

    }

    @After
    public void after(){

    }

    @Test
    public void test_log(){
        Assert.assertEquals("错误消息", "预期的值", "实际的值");
    }
}

File: junit-service/src/main/scala/com/learn/scala/TestDemo.scala
package com.learn.scala

class TestDemo extends UnitSpec {
  override protected def beforeAll(): Unit = {}

  override protected def afterAll(): Unit = {}

  test("base test") {
    assertThrows[ArithmeticException](m1())
    assertResult("预期的值")("实际的值")
    assert("tom" == "tom")
  }

  def m1(): Unit = {
    val num = 10 / 0
  }

  test("exception") {
    try {
      val num = 10 / 0
    } catch {
      case ex: ArithmeticException => throw new Exception("抛出异常")
      case ex: Exception => println("其他异常")
    } finally {
      println("关闭资源")
    }
  }
}
